File: app/src/main/java/com/asmat/rolando/bakingapp/db/ShoppingListDao.java
package com.asmat.rolando.bakingapp.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

/**
 * Created by rolandoasmat on 8/14/17.
 */


@Dao
public interface ShoppingListDao {

    @Query("SELECT * FROM shopping_list")
    List<ShoppingListIngredient> getAllIngredients();

    @Query("SELECT * FROM shopping_list WHERE recipe LIKE :recipeName")
    List<ShoppingListIngredient> getAllIngredientsOfRecipe(String recipeName);

    @Query("SELECT * FROM shopping_list WHERE recipe LIKE :recipeName AND ingredient LIKE :ingredientName LIMIT 1")
    ShoppingListIngredient findIngredient(String recipeName,  String ingredientName);

    @Insert(
            onConflict = 1
    )
    void insert( ShoppingListIngredient var1);

    @Insert(
            onConflict = 1
    )
    void insertAll(List<ShoppingListIngredient> var1);

    @Delete
    void delete(ShoppingListIngredient var1);
}

File: app/src/main/java/com/asmat/rolando/bakingapp/db/ShoppingListDao.kt
package com.asmat.rolando.bakingapp.db

import android.arch.persistence.room.*


/**
 * Created by rolandoasmat on 7/16/17.
 */

@Dao
interface ShoppingListDao {

    @Query("SELECT * FROM shopping_list")
    fun getAllIngredients(): List<ShoppingListIngredient>

    @Query("SELECT * FROM shopping_list WHERE recipe LIKE :arg0")
    fun getAllIngredientsOfRecipe(recipe:String): List<ShoppingListIngredient>

    @Query("SELECT * FROM shopping_list WHERE recipe LIKE :arg0 AND ingredient LIKE :arg1 LIMIT 1")
    fun findIngredient(recipe: String, ingredient: String): ShoppingListIngredient

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(shoppingListIngredient: ShoppingListIngredient)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(ingredients: List<ShoppingListIngredient>)

    @Delete
    fun delete(ingredient: ShoppingListIngredient)
}